cmd/web-handlers.go

```diff
@@ -1852,6 +1861,81 @@ func presignedGet(host, bucket, object string, expiry int64, creds auth.Credenti
 	return host + s3utils.EncodePath(path) + "?" + queryStr + "&" + xhttp.AmzSignature + "=" + signature
 }
 
+// OpenIDConfigResp - OpenIDConfigResp reply.
+type OpenIDConfigResp struct {
+	OpenIDConfigURL string
+	UIVersion       string `json:"uiVersion"`
+}
+
+// OpenIDConfig - get OpenID config.
+func (web *webAPIHandlers) OpenIDConfig(r *http.Request, args *WebGenericArgs, reply *OpenIDConfigResp) error {
+	reply.OpenIDConfigURL = globalServerConfig.OpenID.ConfigURL
+	reply.UIVersion = browser.UIVersion
+	return nil
+}
+
+// LoginSTSArgs - login arguments.
+type LoginSTSArgs struct {
+	Token string `json:"token" form:"token"`
+}
+
+// LoginSTSRep - login reply.
+type LoginSTSRep struct {
+	Token     string `json:"token"`
+	UIVersion string `json:"uiVersion"`
+}
+
+// LoginSTS - STS user login handler.
+func (web *webAPIHandlers) LoginSTS(r *http.Request, args *LoginSTSArgs, reply *LoginRep) error {
+	ctx := newWebContext(r, args, "webLoginSTS")
+
+	v := url.Values{}
+	v.Set("Action", "AssumeRoleWithWebIdentity")
+	v.Set("WebIdentityToken", args.Token)
+	v.Set("DurationSeconds", fmt.Sprintf("%d", 12*60*60))
```

Comment:
This duration seconds you don't need to set, let this arrive from the WebIdentity provider..

Reply:
Done